### PR TITLE
Adding cloud scenarios for multi drive expansion

### PR DESCRIPTION
**Pull request by lsrinivas-pure**
**What this PR does / why we need it**:
Added below scnarios for pool expansion test
- [x] AddDriveAndNodeReboot - Add cloud drive using drive add and reboot node while it is in progress
- [x] MulPoolsResize - Multiple pools expansion using resize-disk parallely
- [x] MulPoolsAddDisk - Multiple pools expansion using add-disk parallely
- [x] ResizeWithJrnlAndMeta - Pool with metadata and journal expansion
- [x] PoolExpandWhileIOAndPXRestart - pools with repl 2 volume , expand one pool node and restart px on other node
- [x] ResizeNodeMaintenanceCycle - expansion using resize-disk and perform node maintenance cycle
- [x] AddDiskNodeMaintenanceCycle - expansion using add-disk and perform node maintenance cycle
- [x] ResizePoolMaintenanceCycle - expansion using resize-disk and perform pool maintenance cycle
- [x] AddDiskPoolMaintenanceCycle -  expansion using add-disk and perform pool maintenance cycle
- [x] NodeMaintenanceResize - pool expansion using resize-disk when node is in maintenance mode
- [x] NodeMaintenanceModeAddDisk - pool expansion using add-disk when node is in maintenance mode
- [x] ResizeNodeMaintenanceMode - pool expansion using resize-disk then  put node in maintenance mode
- [x] AddDiskNodeMaintenanceMode - pool expansion using add-disk then node in maintenance mode
- [x] PoolMaintenanceModeResize - pool expansion using resize-disk when pool is in maintenance mode
- [x] PoolMaintenanceModeAddDisk -  pool expansion using add-disk when pool is in maintenance mode
- [x] ResizePoolMaintenanceMode - pool expansion using resize-disk then put pool  in maintenance mode
- [x] AddDiskPoolMaintenanceMode - pool expansion using add-disk then put  pool in maintenance mode
- [x] PXRestartResize - Restart PX and trigger pool expansion using resize-disk
- [x] PXRestartAddDisk - Restart PX and trigger pool expansion using add-disk
- [x] PoolExpandPendingUntilVolClean - Expand pool should wait until volume gets clean

**Which issue(s) this PR fixes** (optional)
Closes #PTX-13767,PTX-13769,PTX-13786,PTX-13797,PTX-13780

**Special notes for your reviewer**:
http://aetos.pwx.purestorage.com/resultSet/testSetID/61712 
http://aetos.pwx.purestorage.com/resultSet/testSetID/62012
http://aetos.pwx.purestorage.com/resultSet/testSetID/63391
[AddDriveAndNodeReboot.log](https://github.com/portworx/torpedo/files/10371277/AddDriveAndNodeReboot.log)

[PoolExpandWhileIOAndPXRestart.log](https://github.com/portworx/torpedo/files/10371278/PoolExpandWhileIOAndPXRestart.log)

http://aetos.pwx.purestorage.com/resultSet/testSetID/63865
[AddDiskNodeMaintenanceCycle.log](https://github.com/portworx/torpedo/files/10373678/AddDiskNodeMaintenanceCycle.log)

http://aetos.pwx.purestorage.com/resultSet/testSetID/64038
[ResizePoolMaintenanceCycle.log](https://github.com/portworx/torpedo/files/10375770/ResizePoolMaintenanceCycle.log)
http://aetos.pwx.purestorage.com/resultSet/testSetID/64060 
[AddDiskMaintenanceMode.log](https://github.com/portworx/torpedo/files/10375995/AddDiskMaintenanceMode.log)

http://aetos.pwx.purestorage.com/resultSet/testSetID/64813
[PoolMaintenanceModeResize.log](https://github.com/portworx/torpedo/files/10380676/PoolMaintenanceModeResize.log)

http://aetos.pwx.purestorage.com/resultSet/testSetID/65319 
[PoolExpandPendingUntilVolClean.log](https://github.com/portworx/torpedo/files/10386473/PoolExpandPendingUntilVolClean.log)

### CODE REVIEW

File: drivers/volume/portworx/portworx.go
Comment (nikolaypopov) [inline drivers/volume/portworx/portworx.go:641]: Can we improve the logging a bit and make it less complex and more readable. Don't need to give the whole node block, I think just the name will be more readable for debugging purposes, you could maybe add `n.NodeId` if you think its necessary
`log.Warnf("Unable to check if node [%s] is metadata node, Err: %v", *n.Name, err)`

File: drivers/volume/portworx/portworx.go
Comment (nikolaypopov) [inline drivers/volume/portworx/portworx.go:1050]: Don't `log.Errorf()`, just return and lets add a node name, will be easier to debug
`return fmt.Errorf("error when entering pool maintenance on node [%s], Err: %v", n.Name, err)`

File: drivers/volume/portworx/portworx.go
Comment (nikolaypopov) [inline drivers/volume/portworx/portworx.go:1067]: Don't `log.Errorf()`, just return and lets add a node name, will be easier to debug
`return fmt.Errorf("error when exiting pool maintenance on node [%s], Err: %v", n.Name, err)

File: drivers/volume/portworx/portworx.go
Comment (nikolaypopov) [reply drivers/volume/portworx/portworx.go:641]: Also why are we not returning here? Why warning? I think we should return on error, no?

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:1875]: Maybe add the UUID that is used to get the StoragePool, would be good for debugging purposes
`log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)`

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2347]: Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2503]: Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2570]: Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2650]: Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2733]: Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2813]: Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2893]: Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:3072]: Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

**Comment by nikolaypopov** [comment]
We have a lot of issue with Torpedo logging, I think from now on to make it easier to debug, we should make it more verbose.
For example:
`If you are creating pod, its good to mention pod.Name, pod.Namespace and node.Name, if you know where it gets provisioned. If pod creation fails, lets mention in returning error pod.Name, pod.Namespace and node.Name.`

Above is just example, but lets try to log more, just saying I am creating pod doesn't give us much info when something fails.

File: drivers/volume/portworx/portworx.go
Comment (lsrinivas-pure) [reply drivers/volume/portworx/portworx.go:641]: Fixed the node name logging

it is update method, running as pre-check, it fails to check the metadatanode value  sometime and ticket PTX-2445 is raised for the same. Do not want to block the tests due to this. Will address this along with PTX-2445

File: drivers/volume/portworx/portworx.go
Comment (lsrinivas-pure) [reply drivers/volume/portworx/portworx.go:1067]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:1875]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:2503]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:2570]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:2733]: fixd

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:2813]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:2893]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:3072]: fixed

File: drivers/volume/portworx/portworx.go
Comment (lsrinivas-pure) [reply drivers/volume/portworx/portworx.go:1050]: fixed

**Comment by lsrinivas-pure** [comment]
> We have a lot of issue with Torpedo logging, I think from now on to make it easier to debug, we should make it more verbose. For example: `If you are creating pod, its good to mention pod.Name, pod.Namespace and node.Name, if you know where it gets provisioned. If pod creation fails, lets mention in returning error pod.Name, pod.Namespace and node.Name. Don't need to print the whole pod object, as it is too much to read, but instead print basic things that helps members to debug the job`
> 
> Above is just example, but lets try to log more, just saying I am `creating pod` doesn't give us much info when something fails.

sure

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:972]: I would rename this function to `getPoolLastOperation()` since that is what its returning

File: drivers/volume/common.go
Comment (nikolaypopov) [inline drivers/volume/common.go:234]: nit: `on the given node`

File: drivers/volume/common.go
Comment (nikolaypopov) [inline drivers/volume/common.go:242]: nit: `on the given node`

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:1754]: nit: remove extra space

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:1782]: nit: remove extra space

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:1864]: nit: remove extra space

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:1878]: nit: remove extra space

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2555]: nit: remove extra space

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:2631]: nit: remove extra space

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:4572]: nit: remove extra spaces

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:4187]: remove `ValidateApplications(contexts)` if you not using it here

File: tests/basic/storage_pool_test.go
Comment (nikolaypopov) [inline tests/basic/storage_pool_test.go:1850]: nit: remove extra space

**Review by nikolaypopov — APPROVED** [review]
Please fixed minor comments and try to get maybe another review from somebody who is more familiar with these tests you are adding before merging. I am not quite familiar with these multi drive expansion tests and what exactly is required to be automated here, but from code perspective looks ok to me.

File: drivers/volume/common.go
Comment (lsrinivas-pure) [reply drivers/volume/common.go:234]: fixed

File: drivers/volume/common.go
Comment (lsrinivas-pure) [reply drivers/volume/common.go:242]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:972]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:1754]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:1782]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:1850]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:1878]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:2555]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:2631]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:4187]: fixed

File: tests/basic/storage_pool_test.go
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go:4572]: fixed

**Pull request merged**
